core/store/ChainStore/ChainStore.go

```diff
@@ -650,20 +640,49 @@ func (bd *ChainStore) persist(b *Block) error {
 		}
 
 		if b.Transactions[i].TxType == tx.IssueAsset {
-			results := b.Transactions[i].GetMergedAssetIDValueFromOutputs()
+			quantities = b.Transactions[i].GetMergedAssetIDValueFromOutputs()
+		}
 
-			for assetId, value := range results {
-				if _, ok := quantities[assetId]; !ok {
-					quantities[assetId] += value
-				} else {
-					quantities[assetId] = value
-				}
+		for index := 0; index < len(b.Transactions[i].Outputs); index++ {
+			output := b.Transactions[i].Outputs[index]
+			programHash := output.ProgramHash
+			assetId := output.AssetID
+			accountState, err := bd.GetAccount(programHash)
```

Comment:
accountState should first take an existance check from accounts variable, before read from Database .  to prevent data override risk.